storage/src/main/java/tech/pegasys/teku/storage/store/StoreTransactionUpdates.java

```diff
@@ -98,6 +98,8 @@ public void applyToStore(final Store store) {
         (root) -> {
           store.blocks.remove(root);
           store.states.remove(root);
+          store.checkpointStates.removeIf(
```

Comment:
Hmm, removing by slot would leave around states that are based on a block which didn't wind up in the finalized chain but then we processed empty slots into the current epoch.  Current version potentially removes states which are based off the latest finalized block.  So I think we ultimately want to prune by root but ensure we keep states with the latest finalized root.

Reply:
Confirmed that we never prune the finalized block root (which makes sense since we need it around for a lot of things). So I think this works out exactly how we want it to.